resource/src/main/java/za/co/sfy/lucid/sams/resource/AbstractLucidSAMSResource.java

```diff
@@ -0,0 +1,77 @@
+package za.co.sfy.lucid.sams.resource;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import za.co.sfy.lucid.sams.domain.exception.LucidSamsExecutionException;
+import za.co.sfy.lucid.sams.resource.connection.DatabaseConnectionManager;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+
+/**
+ * @author muzim
+ */
+
+public abstract class AbstractLucidSAMSResource {
+
+    private final DatabaseConnectionManager databaseConnectionManager;
+
+    @Autowired
+    public AbstractLucidSAMSResource(final DatabaseConnectionManager databaseConnectionManager) {
+        this.databaseConnectionManager = databaseConnectionManager;
+    }
+
+    /**
+     * Saves Lucid-SAMS POJOs that return a generated id e.g. {@link za.co.sfy.sams.lucid.schema.GeneralInfo}
+     * @param object
+     * @param iLucidSamsResource
+     * @return
+     * @throws LucidSamsExecutionException
+     */
+    public Long save(Object object, ILucidSAMSResource iLucidSamsResource) throws LucidSamsExecutionException {
+
+        Connection databaseConnection = databaseConnectionManager.createDatabaseConnection();
+        String tableName = iLucidSamsResource.getTABLE_NAME();
+        try {
+            PreparedStatement preparedStatement = iLucidSamsResource.getPreparedStatement(databaseConnection, object);
+            int result = preparedStatement.executeUpdate();
+            if (result == 0) {
+                throw new SQLException("'" + tableName + "' insertion failed , no rows were affected");
```

Comment:
Please make sure that you dont have a space before and after the comma so instead: insertion failed, no rows were affected